src/components/CallView/shared/Video.vue

```diff
@@ -180,6 +185,41 @@ export default {
 			return this.isNotConnected && this.model.attributes.connectionState !== ConnectionState.FAILED_NO_RESTART
 		},
 
+		isDisconnected() {
+			return this.model.attributes.connectionState !== ConnectionState.NEW && this.model.attributes.connectionState !== ConnectionState.CHECKING
+				&& this.model.attributes.connectionState !== ConnectionState.CONNECTED && this.model.attributes.connectionState !== ConnectionState.COMPLETED
+		},
+
+		/**
+		 * Whether the connection to the participant is being tried again.
+		 *
+		 * The initial connection to the participant is excluded.
+		 *
+		 * A "failed" connection state will trigger a reconnection, but that may
+		 * not immediately change the "negotiating" or "connecting" attributes
+		 * (for example, while the new offer requested to the HPB was not
+		 * received yet). Similarly both "negotiating" and "connecting" need to
+		 * be checked, as the negotiation will start before the connection
+		 * attempt is started.
+		 */
+		isReconnecting() {
+			return this.model.attributes.connectionState === ConnectionState.FAILED
+				|| (!this.model.attributes.initialConnection
+					&& (this.model.attributes.negotiating || this.model.attributes.connecting))
+		},
+
+		connectionMessage() {
+			if (this.isReconnecting) {
+				return t('spreed', 'Connection problems. Trying to reconnect …')
+			}
+
+			if (this.isDisconnected) {
+				return t('spreed', 'Connection problems …')
```

Comment:
On sending or receiving side?
Should indicate it if possible, so people know whether it's their fault?

Reply:
What does this mean? Something I need to fix? The other end needs to reload? It will retry in x?

Reply:
> On sending or receiving side? Should indicate it if possible, so people know whether it's their fault?

As far as I know it is not possible to know that from the RTCPeerConnection API :-( It only notifies about disconnections, but not whether you can not receive or the other end can not send.

I do not know if that could be detected even with the connection stats; it may not be possible to know that you are able to receive just fine, but the other end is not sending anything (kind of _is it down for everyone or just me?_ situation).

> What does this mean? Something I need to fix? The other end needs to reload? It will retry in x?

I used a short message just to ensure that it will fit in the participant area, but fine by me to use a larger message, I have no preference.

Reply:
> I used a short message just to ensure that it will fit in the participant area, but fine by me to use a larger message, I have no preference.

Doesn't need to be much longer, just needs to say if the user has to do something or not. Can also be `Connection ended. Trying to reconnect …` if that is true and it will reconnect. Otherwise `Connection ended. Reload page to fix it :(` or whatever is to be done by the user.

Reply:
If the HPB is not used and the reconnection has failed five times in a row there will be no more reconnection attempts. This has been there since forever, probably to stop reconnecting endlessly if the connection needs a TURN server but there is none available. In that case there would be nothing that can be done by the user (other than asking the admin to setup a TURN server, for which there is already a message), so maybe in that case the message should be just `Connection can not be established`.

Other than that if a connection fails it should automatically reconnect without the user doing anything. In theory, at least; in some cases the reconnection may never end due to bugs :shrug: (like without the HPB if the Internet cable is unplugged and plugged again after a while).

Reply:
So this can happen after a connection was established as well as when it never was established?

Let's try to be specific then:
* Was connected at some time and HPB is there so it's retrying: `Connection lost. Trying to reconnect …`
* Was connected at some time and NO HPB is there but before 5 fails: `Connection lost. Trying to reconnect …`
* Was connected at some time and NO HPB is there but after 5 fails: `Connection lost and could not be re-established …`
* Was never connected: `Connection could not be established …`

?